src/blocks/author-profile/edit.js

```diff
@@ -0,0 +1,278 @@
+/**
+ * WordPress dependencies
+ */
+import apiFetch from '@wordpress/api-fetch';
+import { InspectorControls } from '@wordpress/block-editor';
+import {
+	Button,
+	ExternalLink,
+	Notice,
+	PanelBody,
+	PanelRow,
+	Placeholder,
+	Spinner,
+	RangeControl,
+	ToggleControl,
+} from '@wordpress/components';
+import { useEffect, useState } from '@wordpress/element';
+import { decodeEntities } from '@wordpress/html-entities';
+import { Icon, people } from '@wordpress/icons';
+import { __, sprintf } from '@wordpress/i18n';
+import { addQueryArgs } from '@wordpress/url';
+
+/**
+ * External dependencies
+ */
+import { AutocompleteWithSuggestions } from 'newspack-components';
+
+export default ( { attributes, setAttributes } ) => {
+	const [ author, setAuthor ] = useState( null );
+	const [ error, setError ] = useState( null );
+	const [ isLoading, setIsLoading ] = useState( false );
+	const [ maxItemsToSuggest, setMaxItemsToSuggest ] = useState( 0 );
+	const {
+		authorId,
+		showBio,
+		showSocial,
+		showEmail,
+		showArchiveLink,
+		showAvatar,
+		avatarBorderRadius,
+	} = attributes;
+
+	useEffect( () => {
+		if ( 0 !== authorId ) {
+			getAuthorById();
+		}
+	}, [ authorId ] );
+
+	const getAuthorById = async () => {
+		setError( null );
+		setIsLoading( true );
+		try {
+			const response = await apiFetch( {
+				path: addQueryArgs( '/newspack-blocks/v1/authors', {
+					author_id: authorId,
+					fields: 'id,name,bio,email,social,avatar,url',
+				} ),
+			} );
+
+			const _author = response.pop();
+
+			if ( ! _author ) {
+				throw sprintf(
+					__( 'No authors or guest authors found for ID %s.', 'newspack-blocks' ),
+					authorId
+				);
+			}
+			setAuthor( _author );
+		} catch ( e ) {
+			setError(
+				e.message ||
+					e ||
+					sprintf(
+						__( 'No authors or guest authors found for ID %s.', 'newspack-blocks' ),
+						authorId
+					)
+			);
+		}
+		setIsLoading( false );
+	};
+
+	// Combine social links and email, which are shown together.
+	const socialLinks = showSocial && author && author.social ? { ...author.social } : {};
```

Comment:
Nit: `{...obj}` is same as `obj`:
```suggestion
	const socialLinks = ( showSocial && author && author.social ) || {};
```

Reply:
I was maybe a bit overzealous when it comes to not mutating the `author` state object, but your suggestion doesn't seem to cause any issues as far as I can tell. Updated in d7e4a91.